examples/prom/grad_div_global_rom.cpp

```diff
@@ -0,0 +1,569 @@
+        //     assembly case, use Jacobi preconditioning.
+        Solver *prec = NULL;
+        CGSolver *pcg = new CGSolver(MPI_COMM_WORLD);
+        pcg->SetOperator(*A);
+        pcg->SetRelTol(1e-12);
+        pcg->SetMaxIter(2000);
+        pcg->SetPrintLevel(1);
+        if (hybridization) {
+            prec = new HypreBoomerAMG(*A.As<HypreParMatrix>());
+        }
+        else if (pa) {
+            prec = new OperatorJacobiSmoother(a, ess_tdof_list);
+        }
+        else
+        {
+            ParFiniteElementSpace *prec_fespace =
+                (a.StaticCondensationIsEnabled() ? a.SCParFESpace() : &fespace);
+            if (dim == 2)   {
+                prec = new HypreAMS(*A.As<HypreParMatrix>(), prec_fespace);
+            }
+            else            {
+                prec = new HypreADS(*A.As<HypreParMatrix>(), prec_fespace);
+            }
+        }
+        pcg->SetPreconditioner(*prec);
+        solveTimer.Start();
+        pcg->Mult(B, X);
+        solveTimer.Stop();
+        delete pcg;
+        delete prec;
+
+        // 18. take and write snapshot for ROM
+        if (offline)
+        {
+            generator->takeSample(X.GetData(), 0.0, 0.01);
+            generator->writeSnapshot();
+            delete generator;
+            delete options;
+        }
+    }
+
+    // 19. The online phase
+    if (online) {
+        // 20. read the reduced basis
+        assembleTimer.Start();
+        CAROM::BasisReader reader(basisName);
+        spatialbasis = reader.getSpatialBasis(0.0);
+        numRowRB = spatialbasis->numRows();
+        numColumnRB = spatialbasis->numColumns();
+        if (myid == 0) printf("spatial basis dimension is %d x %d\n", numRowRB,
+                                  numColumnRB);
+
+        // 21. form inverse ROM operator
+        CAROM::Matrix invReducedA(numColumnRB, numColumnRB, false);
+        ComputeCtAB(*A.As<HypreParMatrix>(), *spatialbasis, *spatialbasis, invReducedA);
+        invReducedA.inverse();
+
+        CAROM::Vector B_carom(B.GetData(), B.Size(), true, false);
+        CAROM::Vector X_carom(X.GetData(), X.Size(), true, false);
+        CAROM::Vector *reducedRHS = spatialbasis->transposeMult(&B_carom);
+        CAROM::Vector reducedSol(numColumnRB, false);
+        assembleTimer.Stop();
+
+        // 22. solve ROM
+        solveTimer.Start();
+        invReducedA.mult(*reducedRHS, reducedSol);
+        solveTimer.Stop();
+
+        // 23. reconstruct FOM state
+        spatialbasis->mult(reducedSol, X_carom);
+        delete spatialbasis;
+        delete reducedRHS;
+    }
+
+    // 24. Recover the parallel grid function corresponding to X. This is the
+    //     local finite element solution on each processor.
+    a.RecoverFEMSolution(X, *b, x);
+
+    // 25. Compute and print the L^2 norm of the error.
+    {
+        double error = x.ComputeL2Error(F);
+        if (myid == 0)
+        {
+            cout << "\n|| F_h - F ||_{L^2} = " << error << '\n' << endl;
```

Comment:
Since there are 2 types of error output, it would be helpful to write something like this here: `Absolute error in L^2 norm with respect to the exact solution`.